frontend/lib/wallets/Coinbase.ts

```diff
@@ -0,0 +1,84 @@
+import CoinbaseWalletSDK, { CoinbaseWalletProvider } from '@coinbase/wallet-sdk';
+import { ethers } from 'ethers';
+import {
+    getChainIdByNetworkType,
+    getDecimalChainIdByNetworkType,
+    getNetworkConfigByType,
+    getNetworkTypeByChainId,
+} from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import CoinbaseLogo from '~/assets/icons/wallets/coinbase.svg';
+import AbstractWallet from '~/lib/wallets/AbstractWallet';
+
+export default class Coinbase extends AbstractWallet {
+    public static title = 'Coinbase Wallet';
+    public static icon = CoinbaseLogo;
+    public static downloadUrl = 'https://www.coinbase.com/wallet/downloads';
+
+    private static provider?: CoinbaseWalletProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    getProvider(network: string): ethers.providers.JsonRpcProvider {
+        const rpcUrl = getNetworkConfigByType(network).url;
+        if (!Coinbase.provider) {
+            const walletSdk = new CoinbaseWalletSDK({ appName: process.env.SITE_TITLE || '' });
+            Coinbase.provider = walletSdk.makeWeb3Provider(rpcUrl);
```

Comment:
Btw, what RPC url _network_ it expects here? Any, on which the website is? Or the url of the network on which the wallet is? Then, how do we get the network which network the wallet is currently on?

On a bigger picture, with metamask for example, we first connect the wallet on whatever network the metamask is and only then we suggest network change (in our UI) in case there is mismatch

Reply:
Would you please still answer the question? 🙂

Reply:
> because we cannot obtain the wallet's network unless we connect it

Well, I think this only depends on the API of the wallet, so it can be possible

Anyway, I would open an issue regarding this for the `WalletConnect`: to provide correct rpc url during connection and after the network switch